src/main/java/games/strategy/ui/SwingComponents.java

```diff
@@ -66,15 +66,15 @@
    * Convenience method to create {@code AbstractAction} instances using lambdas.
    * Example usage:
    * <code><pre>
-   *   AbstractAction action = SwingComponents.newAbstractionAction(() -> {
+   *   AbstractAction action = SwingComponents.newAbstractAction("Something", () -> {
    *    if (value) {
    *      doSomething();
    *    }
    *   });
    * </pre></code>
    */
-  public static AbstractAction newAbstractAction(final Runnable action) {
-    return new AbstractAction() {
+  public static AbstractAction newAbstractAction(final String name, final Runnable action) {
```

Comment:
Is the `name` parameter value ignored or does it overwrite the buttons' name?  Would be nice to see that answered in the javadoc. Though this makes me wonder, `SwingComponents` was supposed to be *the* API Swing was not, instead we are seemingly just wrapping a crappy Swing API with one that allows for lambda's. `JButtonBuilder` solves these problems, my suggestion is to remove `newAbstractAction` completely from `SwingComponents`